tutel/jit_kernels/gating.py

```diff
@@ -20,8 +20,8 @@ def torch_cumsum(mask1):
     return torch_cumsum
 
   global cumsum_kernels
-  if samples in cumsum_kernels:
-    return cumsum_kernels[samples]
+  if (samples, global_experts) in cumsum_kernels:
+    return cumsum_kernels[(samples, global_experts)]
 
```

Comment:
How about: https://github.com/microsoft/tutel/blob/0ccb9bd07b1e4b5cfb41a1210cdf652c2742a11b/tutel/jit_kernels/gating.py#L73

Reply:
Sorry missed that line. Updated.